script/oltpbenchmark/website/run.py

```diff
@@ -0,0 +1,198 @@
+from xml.etree import ElementTree
+from xml.etree.ElementTree import Element
+
+
+OLTPBENCH_DIR = '/home/wanshenl/oltpbench/'
+OLTPBENCH_CONFIG_DIR = os.path.join(OLTPBENCH_DIR, 'config')
+SCRIPT_DIR = '/home/wanshenl/scripts'
+PELOTON_CONFIG_DIR = '/home/wanshenl/scripts/oltpconfig/'
+POSTGRES_CONFIG_DIR = '/home/wanshenl/scripts/oltpconfig_pg/'
+PELOTON_BASE_DIR = '/home/wanshenl/peloton'
+
+BENCHMARKS = [
+    # 'auctionmark',
+    # 'chbenchmark',
+    # 'db2tpcc',
+    # 'epinions',
+    # 'jpab',
+    # 'linkbench',
+    # 'mstpcc',
+    # 'mysmixed',
+    # 'noop',
+    # 'oratpcc',
+    # 'pgmixed',
+    # 'pgtpcc',
+    # 'pgwiki',
+    # 'pgycsb',
+    # 'resourcestresser',
+    # 'seats',
+    # 'sibench',
+    # 'smallbank',
+    'tatp',
+    'tpcc',
+    # 'tpch',
+    # 'twitter',
+    # 'voter',
+    # 'wikipedia',
+    'ycsb'
+]
+
+
+
+def config_xml_file(host, port, benchmark,
+                    transaction, scalefactor, terminals,
+                    time, upload_code, upload_url,
+                    config_in, config_out, username, password):
+    config_path = os.path.join(config_in, 'sample_{}_config.xml'.format(benchmark))
+    xml = ElementTree.parse(config_path)
+    root = xml.getroot()
+    root.find("dbtype").text = "postgres"
+    root.find("driver").text = "org.postgresql.Driver"
+    root.find("DBUrl").text = "jdbc:postgresql://{0}:{1}/{2}".format(host, port, benchmark)
+    root.find("username").text = username
+    root.find("password").text = password
+    uploadCodeElem = Element("uploadCode")
+    uploadCodeElem.text = upload_code.decode('utf-8')
+    root.append(uploadCodeElem)
+    uploadUrlElem = Element("uploadUrl")
+    uploadUrlElem.text = upload_url.decode('utf-8')
+    root.append(uploadUrlElem)
+    if 'isolation' not in root:
+        isoElem = Element("isolation")
+        isoElem.text = str(transaction)
+        root.append(isoElem)
+    root.find("isolation").text = str(transaction)
+    root.find("scalefactor").text = str(scalefactor)
+    root.find("terminals").text = str(terminals)
+    for work in root.find('works').findall('work'):
+        if 'time' not in work:
+            timeElem = Element("time")
+            timeElem.text = str(time)
+            work.append(timeElem)
+        work.find('time').text = str(time)
+        work.find('rate').text = "unlimited"
+    outfile = config_out + benchmark + '_config.xml'
+    xml.write(outfile, encoding='UTF-8')
+    print("Wrote {}".format(outfile))
+
+
+def run_commands(commands):
+    for command in commands:
+        x = command.split(' ')
+        if x[0] == 'cd':
+            os.chdir(x[1])
+        else:
+            os.system(command)
+
+
+def peloton_build():
+    peloton_path = PELOTON_BASE_DIR
+    install_script_path = os.path.join(peloton_path, 'script', 'installation')
+    build_path = os.path.join(peloton_path, 'build')
+    bin_path = os.path.join(peloton_path, 'bin')
+
+    commands = [
+        'cd ' + peloton_path,
```

Comment:
I would convert the command list into a set of small functions (class methods).  The directories where things run shouldn't be explicitly stated for each step.
e.g.
def run_cmake_(self):
  os.chdir(self.build_path)
  ret_val = subprocess.call(... some command)
  check return value and report error if needed